drafts/str4d-stratum/draft1.rst

```diff
@@ -0,0 +1,256 @@
+  reconnecting.
+
+If ``client.reconnect`` is sent with empty parameters, the miner SHOULD
+reconnect to the same host and port it is currently connected to.
+
+``mining.suggest_target()``
+---------------------------
+
+Request (optional)::
+
+    {"id": 3, "method": "mining.suggest_target", "params": ["TARGET"]}\n
+
+- ``TARGET`` (hex): The target suggested by the miner for the next received job
+  and all subsequent jobs (until the next time this message is sent).
+
+The server SHOULD reply with ``mining.set_target``. The server MAY set the
+result id equal to the request id.
+
+
+Motivation
+==========
+
+Many existing cryptocurrency miners and pools use the original Stratum protocol
+for communication, in situations where the miner does not require any control
+over what they mine (for example, a miner connected to a local P2Pool node).
+However, the protocol is very specific to Bitcoin, in that it makes assumptions
+about the block header format, and the available nonce space. Zcash has made
+changes that invalidate these assumptions.
+
+Having a formal specification for a Zcash-compatible Stratum-style mining
+protocol means that existing pool operators and miner authors can quickly and
+easily migrate their frameworks to the Zcash network, with no ambiguity about
+interoperability.
+
+
+Rationale
+=========
+
+Why use the 256-bit target instead of a numerical difficulty?
+
+- There is no protocol ambiguity when using a target. A server can pick a
+  specific target (by whatever algorithm), and enforce it cleanly on submitted
+  jobs.
+
+  - A numerical difficulty must be converted into a target by miners, which adds
+    unnecessary complexity, results in a loss of precision, and leaves ambiguity
+    over the conversion and the validity of resulting submissions.
+
+- The minimum numerical difficulty in Bitcoin's Stratum protocol is 1, which
+  corresponds to ``powLimit``. This makes it harder to test miners and servers.
+  A target can represent difficulties lower than the minimum.
+
+Does a 256-bit target waste bandwidth?
+
+- The target is generally not set as often as solutions are submitted, so any
+  effect is minimal.
+
+- Zcash's proof-of-work, Equihash, is much slower than Bitcoin's, so any latency
+  caused by the size of the target is minimal compared to the overall solver
+  time.
+
+- For the current Equihash parameters (200/9), the Equihash solution dominates
+  bandwidth usage.
+
+
+Reference Implementation
+========================
+
+- `str4d's standalone miner`_
+
+.. _`str4d's standalone miner`: https://github.com/str4d/zcash/tree/standalone-miner
+
+
+Acknowledgements
+================
+
+Thanks to:
+
+- 5a1t for the initial brainstorming session.
+
+- Daira Hopwood for her input on API selection and design.
+
+- Marek Palatinus (slush) and his colleagues for their refinements, suggestions, and
+  robust discussion.
+
+This ZIP was edited by [TODO: fill in]
+
+
+References
+==========
+
```

Comment:
Put the citation definitions close to their first use; they'll be moved here. So this section should be blank with an rST comment saying that.